consumer.go

```diff
@@ -220,8 +222,35 @@ func WithLogger(logger *slog.Logger) ConsumerOption {
 	}
 }
 
+// MetadataFilter is a filter for metadata. Right now support only direct matching of key/value
+type (
+	MetadataOperation string
+
+	MetadataFilter struct {
+		Key       string
+		Operation MetadataOperation
+		Value     string
+	}
+)
+
+const (
+	OpEqual    MetadataOperation = "="
+	OpNotEqual MetadataOperation = "<>"
+)
+
+func WithMetadataFilter(filter *MetadataFilter) ConsumerOption {
+	return func(c *consumerConfig) {
+		filters := c.MetadataFilters
+		if filter == nil {
+			filters = make([]MetadataFilter, 0, 1)
+		}
+
+		c.MetadataFilters = append(filters, *filter)
+	}
+}
+
 // NewConsumer creates Consumer with proper settings
-func NewConsumer(db *sql.DB, queueName string, handler MessageHandler, opts ...ConsumerOption) (*Consumer, error) {
+func NewConsumer(db *sqlx.DB, queueName string, handler MessageHandler, opts ...ConsumerOption) (*Consumer, error) {
```

Comment:
Can be a problem in a future.  
https://github.com/dataddo/pgq/issues/8

Reply:
Keeping sqlx.DB for Named Param support

Reply:
Hi, just a note I know I've recommended "sqlx" but I've checked it couple days ago because of some other project and development seems to be death (no commits in last couple of years). When you check lib/pq repo they currently recommending -  "For users that require new features or reliable resolution of reported bugs, we recommend using [pgx](https://github.com/jackc/pgx) which is under active development." (at the end of readme). Which has named params but I've never tested/use it. Currently I'm still using sqlx without any issues but "death development" can cause issues in a future